pkg/apis/maistra/v2/security.go

```diff
@@ -235,6 +223,19 @@ type ControlPlaneSecurityConfig struct {
 	TLS *ControlPlaneTLSConfig `json:"tls,omitempty"`
 }
 
+// DataPlaneSecurityConfig is the mutual TLS configuration specific to the
+// control plane.
+type DataPlaneSecurityConfig struct {
+	// Enable mutual TLS by default.
+	// .Values.global.mtls.enabled
+	MTLS *bool `json:"mtls,omitempty"`
+	// Auto configures the mesh to automatically detect whether or not mutual
+	// TLS is required for a specific connection.
+	// .Values.global.mtls.auto
+	// +optional
+	AutoMTLS *bool `json:"automtls,omitempty"`
```

Comment:
nit: should the json tag be `autoMTLS`?  Not sure what the standard style is for things like this.

Reply:
I tried that it looked horrible. Happy to change if we prefer it, but I though `automtls` and `mtls` are a better fit than `autoMTLS` and `mtls`

Reply:
works for me.  are there any other json tags that you think we should change?  i'd just like to be somewhat consistent (and i struggle with this myself).

Reply:
as discussed on IRC: not really :)